### PR TITLE
Deny unused dependencies

### PR DESCRIPTION
**Pull request by Razz4780**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -56,8 +56,9 @@ categories = ["development-tools", "backend", "devtool"]
 [workspace.dependencies]
 actix-codec = "0.5"
 bincode = { version = "2.0.0-rc.2", features = ["serde"] }
-bytes = "1"
-tokio = { version = "1" }
+bytes = "1.7"
```
Comment (aviramha) [inline Cargo.toml:59]: Why do we prefer to be specific?

File: mirrord/agent/Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ workspace = true
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
-[dependencies]
+[target.'cfg(target_os = "linux")'.dependencies]
```
Comment (aviramha) [inline mirrord/agent/Cargo.toml:21]: Nice!

File: mirrord/auth/src/lib.rs [hunk 3]
```diff
@@ -14,3 +15,13 @@ pub mod credentials;
 pub mod error;
 /// Public/Private key abstraction for serialization and deserialization
 pub mod key_pair;
+
+mod deps_required_for_compilation {
```
Comment (aviramha) [inline mirrord/auth/src/lib.rs:19]: I love this doc!

File: Cargo.toml (hunk 1 above)
Comment (Razz4780) [reply Cargo.toml:59]: Hmm, not sure actually. Just my habit :sweat_smile:

Comment (aviramha) [reply Cargo.toml:59]: I prefer the less specific, as we have lock anyway..

File: mirrord/kube/src/lib.rs [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 
 //! # Features
 //!
 //! ## `incluster`
 //!
-//! Turn this feature on if you want this code to be run from inside a Kubernetes cluster.
-//! It affects the way [`api::kubernetes::KubernetesAPI`] connects to created agents.
-//! From outside of the cluster, [`kube`]s port forwarding is used.
-//! From inside of the cluster, plain TCP connection is made.
+//! Turn this feature on if you want to connect to agent pods from withing the cluster with a plain
```
Comment (meowjesty) [inline mirrord/kube/src/lib.rs:10]:
```suggestion
//! Turn this feature on if you want to connect to agent pods from within the cluster with a plain
```

File: mirrord/layer/Cargo.toml [hunk 5]
```diff
@@ -27,52 +27,46 @@ mirrord-intproxy-protocol = { path = "../intproxy/protocol", features = [
...
 tempfile.workspace = true
-test-cdylib = "*"
-tower.workspace = true
-tokio = { workspace = true, features = ["rt", "net", "macros"] }
 tests = { path = "../../tests" }
-flaky_test = "0.2"
-futures.workspace = true
-actix-codec.workspace = true
-tokio-stream.workspace = true
+test-cdylib = "*"
```
Comment (meowjesty) [inline mirrord/layer/Cargo.toml:68]:
```suggestion
test-cdylib = "1"
```

**Review by meowjesty — APPROVED** [review]
:+1:

**Pull request merged**